presto-main/src/main/java/com/facebook/presto/operator/HashBuilderOperator.java

```diff
@@ -495,6 +499,16 @@ private void finishInput()
         state = State.LOOKUP_SOURCE_BUILT;
     }
 
+    @VisibleForTesting
```

Comment:
I'm a little bit reluctant of adding this type of testing. You can simply append manually constructed page with well know size, and then check if memory accounting correctly reflects that page size in the index. 

Additionally you can have a unit test for the `PagesHash.calculatedPredictedSizeDelta`. For the unit test you can go either way. You can either compare the result with a hardcoded value, or you can run the graph traversal.

Reply:
(y)